fpdf/output.py

```diff
@@ -1785,10 +1811,49 @@ def _finalize_catalog(
         catalog_obj.struct_tree_root = struct_tree_root_obj
         catalog_obj.outlines = outline_dict_obj
         catalog_obj.metadata = xmp_metadata_obj
-        if sig_annotation_obj:
-            flags = SignatureFlag.SIGNATURES_EXIST + SignatureFlag.APPEND_ONLY
+
+        # Collect all form fields from all pages
+        all_form_fields = []
+        for page_obj in fpdf.pages.values():
+            for annot_obj in page_obj.annots:
+                if isinstance(annot_obj, FormField):
+                    all_form_fields.append(annot_obj)
+
+        # Build AcroForm if there are form fields or a signature
+        if all_form_fields or sig_annotation_obj:
+            # Combine signature and form fields
+            acro_fields = []
+            sig_flags = None
+
+            if sig_annotation_obj:
+                acro_fields.append(sig_annotation_obj)
+                sig_flags = SignatureFlag.SIGNATURES_EXIST + SignatureFlag.APPEND_ONLY
+
+            acro_fields.extend(all_form_fields)
+
+            # Build default resources with standard fonts for form fields
+            default_resources = None
+            default_appearance = None
+            if all_form_fields:
+                # /DR dictionary with Helvetica and ZapfDingbats fonts
+                # These are standard PDF fonts that don't require embedding
+                default_resources = (
+                    "<</Font <<"
+                    "/Helv <</Type /Font /Subtype /Type1 /BaseFont /Helvetica /Encoding /WinAnsiEncoding>> "
+                    "/ZaDb <</Type /Font /Subtype /Type1 /BaseFont /ZapfDingbats>>"
+                    ">>>>"
+                )
```

Comment:
I am not sure having a hardcoded resource dictionary with /Helv /ZaDb is the best approach - tapping into the main FPDF resources (or at least the ResourceCatalog) will give you much more choices and flexibility, including user added/TTF fonts.